parrot_controller/include/waypoint.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2016 PolySync
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * @file waypoint.h
+ * @brief Waypoint storage.
+ *
+ */
+
+
+
+
+#ifndef WAYPOINT_H
+#define WAYPOINT_H
+
+
+#include "gui.h"
+
+
+
+
+#define MAX_WAYPOINTS 10
+
+#define REACHED_WAYPOINT_DISTANCE 10
+
+
+
+
+/**
+ * @brief Waypoint.
+ *
+ */
+typedef struct
+{
+    //
+    //
+    double                      x; /*!< Waypoint X coordinate. */
+    //
+    //
+    double                      y; /*!< Waypoint Y coordinate. */
+    //
+    //
+    double                      heading; /*!< Waypoint heading. */
+    //
+    //
+    int                         valid; /*!< Is Waypoint valid */
+} waypoint_s;
+
+
+
+void zero_waypoints( waypoint_s * waypoints );
```

Comment:
c-style calls for constant pointer when possible, same for pointer to constant

also documentation is missing